content/articles/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/index.md

```diff
@@ -0,0 +1,226 @@
+
+- `"outDir": "./dist"` - this an output directory where the output structure of the compiled JavaScript will be saved. In this case, JavaScript code will be saved in a folder called to `dist` in the root directory of the main project.
+
+- `"moduleResolution": "node"` - a module import resolution algorithm that mimics the way the Node.js searches for modules in real-time.
+
+- `"strict": true` - enable all JavaScript strict type-checking options.
+
+- `"esModuleInterop": true` - esModuleInterop allows us to compile ES6 modules to commonjs modules.
+
+- `"exclude":[]` - tells Typescript not to compile specified files or folders. In this case, you can tell the Typescript not to compile the `node_modules` folder since it already installed the JavaScript versions of Node.js library dependencies. I.e.;
+
+```json
+"exclude":[
+    "./node_modules"
+]
+```
+
+Check this [tutorial](https://www.typescriptlang.org/tsconfig) to learn more options that you can include in your tsconfig.json file.
+
+Here is an example of a `tsconfig.json` file option.
+
+```JSON
+{
+  "compilerOptions": {                        
+    "target": "es6",                               
+    "module": "commonjs",                           
+    "outDir": "./dist",                             
+    "rootDir": "./src",                             
+    "strict": true,
+    "moduleResolution": "node",
+    "esModuleInterop": true,                       
+  },
+  "exclude":[
+    "./node_modules"
+  ]
+}
+```
+
+Feel free to add or change this file as needed to configure the Typescript compiler to your application and environment needs.
+
+### Using Typescript with Node.js
+
+Go ahead and create the folder `src` in your project directory. `src` hosts the Typescript files as explained earlier. Inside `src`, create an `index.ts` file and start writing some Typescript code. While writing down Typescript, it is advisable to have a Typescript compiler installed in your project's local dependencies. Run `npm install -D typescript` to do so.
+
+```ts
+function sum (num1:number, num2:number){
+    return num1 + num2;
+}
+console.log(sum(8,4))
+```
+
+Let's see how we can execute the above Typescript using Node.js. Run `tsc` to build Typescript into JavaScript. This will build, compile and output JavaScript code into the path you specified in `"outDir": "./dist"` of the `config.json` file.
+
+![Compiled JavaScript](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/compiled-javascript.png)
+
+Now run the below command to execute the code:
+
+```bash
+node dist/index.js
+``` 
+
+![Results](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/sum-results.png)
+
+Running the above commands every single time to compile and run in a development environment can be annoying. To make this process easier, you need to install a package called `ts-node`.
+
+```bash
+npm install -D ts-node
+```
```

Comment:
This package requires typescript to be installed in the local project directory.